include/drivers/clock_control.h

```diff
@@ -103,9 +103,17 @@ struct clock_control_driver_api {
 
 /**
  * @brief Enable the clock of a sub-system controlled by the device
- * @param dev Pointer to the device structure for the clock controller driver
- * 	instance
- * @param sys A pointer to an opaque data representing the sub-system
+ *
+ * This function is not blocking: the clock may or may not be enabled
```

Comment:
That's actually wrong. See my comment on your issue. 

Common pattern being: functions ending with _async are non blocking, others are.
The documentation was missing at that time, yes. But these were meant to be blocking and should be blocking or then they would be aliases to _async counterpart with a cb set as NULL... so they would be useless.

Reply:
Thanks for responding. I asked you on slack yesterday what you meant but got no answer :).

OK, so then the situation is that all the clock drivers which aren't blocking are buggy. Would you agree?

Reply:
>  functions ending with _async are non blocking, others are.

BTW, I fully agree with this, but nobody who responded so far seemed to, so I am glad we are on the same page.

Reply:
> Would you agree?

I see from your comment on the issue that you do agree. Thanks!

Reply:
> > functions ending with _async are non blocking, others are.
> 
> BTW, I fully agree with this, but nobody who responded so far seemed to, so I am glad we are on the same page.

I also think this should be true, but see https://github.com/zephyrproject-rtos/zephyr/issues/15703#issuecomment-487470785.  "Non-blocking" is subject to conditions, at least for some API.

Reply:
That comment seems to indicate that "non-blocking" is not the same as "queued", which makes sense.

Reply:
> Thanks for responding. I asked you on slack yesterday what you meant but got no answer :).

Sorry I missed the message/notification 

> OK, so then the situation is that all the clock drivers which aren't blocking are buggy. Would you agree?

Yes, we are on same page.